src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -292,7 +292,7 @@ private static function normalizeHeaders(array $headers): array
                 [$name, $values] = explode(':', $values, 2);
                 $values = [ltrim($values)];
             } elseif (!is_iterable($values)) {
-                if (\is_object($values)) {
+                if (\is_object($values) || $values instanceof \Closure) {
```

Comment:
a closure is an object so this doesn't add anything

Reply:
maybe just `!is_string($values)` ?